packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -525,28 +530,35 @@ class BallisticScrollActivity extends ScrollActivity {
   /// The [delegate], [simulation], and [vsync] arguments must not be null.
   BallisticScrollActivity(
     ScrollActivityDelegate delegate,
-    Simulation simulation,
+    ScrollSimulation simulation,
     TickerProvider vsync,
-  ) : super(delegate) {
+  ) : _simulation = simulation, super(delegate) {
     _controller = AnimationController.unbounded(
       debugLabel: kDebugMode ? objectRuntimeType(this, 'BallisticScrollActivity') : null,
       vsync: vsync,
     )
       ..addListener(_tick)
-      ..animateWith(simulation)
+      ..animateWith(_simulation)
        .whenComplete(_end); // won't trigger if we dispose _controller first
   }
 
   late AnimationController _controller;
 
+  final ScrollSimulation _simulation;
+
   @override
   void resetActivity() {
     delegate.goBallistic(velocity);
   }
 
   @override
   void applyNewDimensions() {
-    delegate.goBallistic(velocity);
+    final ScrollPosition position = delegate.scrollPosition;
```

Comment:
Could this change just be implemented in the scroll position, so you do not need to add a position getter to the delegate?